src/cdomains/threadIdDomain.ml

```diff
@@ -39,6 +39,9 @@ sig
 
   val threadenter: t * D.t -> Node.t -> varinfo -> t
   val threadspawn: D.t -> Node.t -> varinfo -> D.t
+
+  (** If it is possible to get a list of unique thread create thus far, get it *)
+  val created: t -> D.t -> (t list) option
```

Comment:
Does this need to be `t list option` or could it just be `t list`, which returns `[]` when it's not possible to get a list?

Reply:
It needs to be `t list option` to distinguish between whether no threads are created yet, or whether such information is not available in the current domain.

https://github.com/goblint/analyzer/blob/e5d63c856d8284cd9fb4a85a33408333ba28a6eb/src/analyses/threadId.ml#L89-L97